File: 백준/Gold/14719. 빗물/빗물.py
h, w = map(int, input().split())

lst = list(map(int, input().split()))
result = 0

for i in range(1, len(lst)-1):
    lmax, rmax = 0, 0
    for j in range(0, i):
        if lst[j] > lmax:
            lmax = lst[j]
    for k in range(i+1, len(lst)):
        if lst[k] > rmax:
            rmax = lst[k]
    
    mValue = min(lmax, rmax)

    value = mValue - lst[i]

    if value > 0:
        result += value

print(result)

File: 백준/Gold/14719. 빗물/빗물.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        
        int h = sc.nextInt();
        int w = sc.nextInt();
        
        int[] lst = new int[w];
        for (int i = 0; i < w; i++) {
            lst[i] = sc.nextInt();
        }
        
        int result = 0;
        
        for (int i = 1; i < w - 1; i++) {
            int lmax = 0, rmax = 0;
            
            for (int j = 0; j < i; j++) {
                if (lst[j] > lmax) {
                    lmax = lst[j];
                }
            }
            
            for (int k = i + 1; k < w; k++) {
                if (lst[k] > rmax) {
                    rmax = lst[k];
                }
            }
            
            int mValue = Math.min(lmax, rmax);
            int value = mValue - lst[i];
            
            if (value > 0) {
                result += value;
            }
        }
        
        System.out.println(result);
    }
}